tests/Parallafka.Tests/Helpers/Wait.cs

```diff
@@ -61,11 +61,14 @@ public static Task UntilAsync(string desiredStateDescription, Func<Task> asserti
 
         public static async Task ForTaskOrTimeoutAsync(Task task, TimeSpan timeout, Action onTimeout)
         {
-            var timeoutTask = Task.Delay(timeout);
-            await Task.WhenAny(task, timeoutTask);
-            if (timeoutTask.IsCompleted && !task.IsCompleted)
+            await Task.WhenAny(task, Task.Delay(timeout));
+            if (!task.IsCompleted)
+            {
+                onTimeout?.Invoke();
+            }
+            else
             {
-                onTimeout.Invoke();
+                await task;
```

Comment:
Great catch! I assumed WhenAny would bubble up exceptions like WhenAll. I guess I use WhenAll a lot more.

Reply:
The WhenAny deal w/ NOT throwing exceptions has caught me enough times that I now remember to do this :)